source/views/stoprint/print-jobs.js

```diff
@@ -0,0 +1,159 @@
+// @flow
+
+import React from 'react'
+import {SectionList} from 'react-native'
+import {connect} from 'react-redux'
+import {type ReduxState} from '../../flux'
+import {updatePrintJobs} from '../../flux/parts/stoprint'
+import {type PrintJob, STOPRINT_HELP_PAGE} from '../../lib/stoprint'
+import {
+	ListRow,
+	ListSeparator,
+	ListSectionHeader,
+	Detail,
+	Title,
+} from '../components/list'
+import type {TopLevelViewPropsType} from '../types'
+import delay from 'delay'
+import openUrl from '../components/open-url'
+import {StoPrintErrorView, StoPrintNoticeView} from './components'
+import groupBy from 'lodash/groupBy'
+import toPairs from 'lodash/toPairs'
+import sortBy from 'lodash/sortBy'
+
+type ReactProps = TopLevelViewPropsType
+
+type ReduxStateProps = {
+	jobs: Array<PrintJob>,
+	error: ?string,
+	loading: boolean,
+	loginState: string,
+}
+
+type ReduxDispatchProps = {
+	updatePrintJobs: () => Promise<any>,
+}
+
+type Props = ReactProps & ReduxDispatchProps & ReduxStateProps
+
+class PrintJobsView extends React.PureComponent<Props> {
+	static navigationOptions = {
+		title: 'Print Jobs',
+		headerBackTitle: 'Jobs',
+	}
+
+	componentDidMount() {
+		this.refresh()
```

Comment:
Have this call `this.fetchData` instead; there's no need to add the 500ms delay unless you're doing PullToRefresh.